jscomp/core/js_record_iter.ml

```diff
@@ -0,0 +1,149 @@
+expression_desc : expression_desc fn;
+for_ident_expression : for_ident_expression fn;
+finish_ident_expression : finish_ident_expression fn;
+statement_desc : statement_desc fn;
+expression : expression fn;
+statement : statement fn;
+variable_declaration : variable_declaration fn;
+string_clause : string_clause fn;
+int_clause : int_clause fn;
+case_clause : case_clause fn;
+block : block fn;
+program : program fn;
+deps_program : deps_program fn
+    }  
+    and 'a fn = iter -> 'a -> unit
+    let iter : iter = {
+     label : label fn  = ( unknown )  ;
+ required_modules : required_modules fn  = ( fun _self arg -> list ((fun _self arg -> _self.module_id _self arg)) _self arg )  ;
+ ident : ident fn  = ( unknown )  ;
+ module_id : module_id fn  = ( fun _self { id = _x0;kind = _x1} -> begin _self.ident _self _x0 end )  ;
+ vident : vident fn  = ( fun _self -> function 
+| Id ( _x0)  -> 
+ begin _self.ident _self _x0 end
+|Qualified ( _x0,_x1)  -> 
+ begin _self.module_id _self _x0 end )  ;
+ exception_ident : exception_ident fn  = ( (fun _self arg -> _self.ident _self arg) )  ;
+ for_ident : for_ident fn  = ( (fun _self arg -> _self.ident _self arg) )  ;
+ for_direction : for_direction fn  = ( unknown )  ;
+ property_map : property_map fn  = ( fun _self arg -> list ((fun _self (_x0,_x1) -> begin _self.expression _self _x1 end)) _self arg )  ;
+ length_object : length_object fn  = ( unknown )  ;
+ expression_desc : expression_desc fn  = ( fun _self -> function 
+| Length ( _x0,_x1)  -> 
+ begin _self.expression _self _x0;_self.length_object _self _x1 end
+|Char_of_int ( _x0)  -> 
+ begin _self.expression _self _x0 end
+|Char_to_int ( _x0)  -> 
+ begin _self.expression _self _x0 end
+|Is_null_or_undefined ( _x0)  -> 
+ begin _self.expression _self _x0 end
+|String_append ( _x0,_x1)  -> 
+ begin _self.expression _self _x0;_self.expression _self _x1 end
+|Bool _ -> ()
+|Typeof ( _x0)  -> 
+ begin _self.expression _self _x0 end
+|Js_not ( _x0)  -> 
+ begin _self.expression _self _x0 end
+|Seq ( _x0,_x1)  -> 
+ begin _self.expression _self _x0;_self.expression _self _x1 end
+|Cond ( _x0,_x1,_x2)  -> 
+ begin _self.expression _self _x0;_self.expression _self _x1;_self.expression _self _x2 end
+|Bin ( _x0,_x1,_x2)  -> 
+ begin _self.expression _self _x1;_self.expression _self _x2 end
+|FlatCall ( _x0,_x1)  -> 
+ begin _self.expression _self _x0;_self.expression _self _x1 end
+|Call ( _x0,_x1,_x2)  -> 
+ begin _self.expression _self _x0;list ((fun _self arg -> _self.expression _self arg)) _self _x1 end
+|String_index ( _x0,_x1)  -> 
+ begin _self.expression _self _x0;_self.expression _self _x1 end
+|Array_index ( _x0,_x1)  -> 
+ begin _self.expression _self _x0;_self.expression _self _x1 end
+|Static_index ( _x0,_x1,_x2)  -> 
+ begin _self.expression _self _x0 end
+|New ( _x0,_x1)  -> 
+ begin _self.expression _self _x0;option (fun _self arg -> list ((fun _self arg -> _self.expression _self arg)) _self arg) _self _x1 end
```

Comment:
this could be simplified too

Reply:
option (list _self.expression)
Note we should also be careful about the arity mismatch

Reply:
simplified as :
```
option (fun _self arg -> list _self.expression _self arg)
```
could be even simplified as 
```
option (list _self.expression)
```
This created a partial application which is not worth it